src/awkward/contents/indexedarray.py

```diff
@@ -663,9 +666,24 @@ def _mergemany(self, others: Sequence[Content]) -> Content:
                 contentlength_so_far += array.length
                 length_so_far += array.length
 
+                # Categoricals may only survive if all contents are categorical
+                if (
+                    parameters is not None
+                    and parameters.get("__array__") == "categorical"
+                ):
+                    parameters = {**parameters}
+                    del parameters["__array__"]
```

Comment:
This handles the case of merging content X against `categorical[Y]`, whereby `Y` is merged against `X`. In this case, we need to recognise that the outer content (`categorical[Y | X]`) is no longer categorical.

Reply:
That's good.